src/Compilers/CSharp/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.cs

```diff
@@ -129,6 +131,23 @@ private static bool IsExcludedFromCodeCoverage(MethodSymbol method)
             return false;
         }
 
+        private BoundExpression GetFileIndices()
+        {
+            if (_documentsBuilder.Count == 0)
+            {
+                // For a compiler generated method that has no 'real' spans, remember the source
+                // document corresponding to the syntax node that is associated with its bound node.
+                _documentsBuilder.Add(GetSourceDocument(_methodBody.Syntax));
+            }
+
+            ImmutableArray<BoundExpression> fileIndices = _documentsBuilder.ToImmutableArray()
+                .SelectAsArray(d => _methodBodyFactory.SourceDocumentIndex(d));
```

Comment:
Consider using overload with extra argument to avoid closure class: `.SelectAsArray((d, f) => f.SourceDocumentIndex(d), _methodBodyFactory)`